src/status.ts

```diff
@@ -2,258 +2,396 @@ import * as vscode from 'vscode';
+    return Boolean(vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 1 && !!this.getText());
+  }
+  protected getTextIcon():string {
+    return '$(folder-active)';
   }
+  protected getTooltipIcon():string {
+    return `CMake: ${this.getTooltipNormal()}\n${this.getTextNormal()}`;
+  }
+}
+
+class CMakeStatus extends Button {
+  settingsName = 'status';
+  command = "cmake.setVariant";
+  tooltip = localize('click.to.select.variant.tooltip', 'Click to select the current build variant');
 
-  /**
-   * The message shown in the primary status button. Tells the user what the
-   * extension is currently up to.
-   */
+  private _buildTypeLabel: string = localize('unconfigured', 'Unconfigured');
   private _statusMessage: string = localize('loading.status', 'Loading...');
-  setStatusMessage(v: string) {
+
+  set buildTypeLabel(v: string) {
+    this._buildTypeLabel = v;
+    this.update();
+  }
+  set statusMessage(v: string) {
     this._statusMessage = v;
-    this._reloadStatusButton();
+    this.update();
   }
-
-  /**
-   * The name of the currently active target to build
-   */
-  private _targetName: string = '';
-  public get targetName(): string { return this._targetName; }
-  public set targetName(v: string) {
-    this._targetName = v;
-    this._targetButton.text = `[${v}]`;
-    this.reloadVisibility();
+  protected getTextNormal() {
+    return `CMake: ${this._buildTypeLabel}: ${this._statusMessage}`;
+  }
+  protected getTextShort() {
+    return `${this._buildTypeLabel}: ${this._statusMessage}`;
   }
+}
 
-  setLaunchTargetName(v: string) {
-    this._launchTargetNameButton.text = v;
-    this._reloadDebugButton();
+class KitSelection extends Button {
+  settingsName = 'kit';
+  command = 'cmake.selectKit';
+  tooltip = localize('click.to.change.kit.tooltip', 'Click to change the active kit');
+
+  protected getTextNormal():string {
+    const text = this.text;
+    if (text === SpecialKits.Unspecified) {
+      return `[${localize('no.active.kit', 'No active kit')}]`;
+    }
+    if (text.length === 0) {
+      return localize('no.kit.selected', 'No Kit Selected');
+    }
+    return text;
+  }
+  protected getTextIcon(): string {
+    return `$(tools)`;
+  }
+  protected getTextShort() {
+    const len = Number(this._config.statusbar.advanced?.kit?.length) || 20;
+    let text = this.getTextNormal();
+    if (len + 3 < text.length) {
+      text = `${text.substr(0, len)}...`;
+    }
+    return text;
+  }
+  protected getTooltipShort(): string {
+    return `${this.getTooltipNormal()}\n${this.getTextNormal()}`;
   }
+  protected getTooltipIcon(): string {
+    return `${this.getTooltipNormal()}\n${this.getTextNormal()}`;
+  }
+}
 
-  private _ctestEnabled: boolean = false;
-  public get ctestEnabled(): boolean { return this._ctestEnabled; }
-  public set ctestEnabled(v: boolean) {
-    this._ctestEnabled = v;
-    setVisible(this._testButton, v);
+class BuildTargetSelectionButton extends Button {
```

Comment:
I think that the two target selection buttons should hide when "icon" mode is set. I consider them more as labels than buttons that do something (even though they do have a shortcut to the command that lets you change the target)

Reply:
added with 798e21c6191ade40c56702bba03f0f1a512f7bdd